### PR TITLE
Adding viridi's veil hexproof mod

### PR DESCRIPTION
**Pull request by Wires77**

### CODE REVIEW

File: Modules/ModParser.lua [hunk 1]
```diff
@@ -3416,6 +3417,9 @@ local count = 0
 --foo:close()
 return function(line, isComb)
 	if not cache[line] then
+		if line:match("you are hexproof") then
```
Comment (PJacek) [inline Modules/ModParser.lua:3420]: What's the purpose of this block?

Comment (Wires77) [reply Modules/ModParser.lua:3420]: I accidentally left it in when debugging

File: Modules/ModParser.lua [hunk 2]
```diff
@@ -2356,6 +2356,7 @@ local specialModList = {
 		mod("AvoidStun", "BASE", 100, { type = "Condition", var = "UsingFlask" }),
 	},
 	["unaffected by curses"] = { mod("CurseEffectOnSelf", "MORE", -100) },
+	["you are hexproof if you have a (%a+) (%a+) in (%a+) slot"] = function(_, rarity, item, slot) return { mod("AvoidCurse", "BASE", 100, { type = "Condition", var = rarity:gsub("^%l", string.upper).."ItemIn"..item:gsub("^%l", string.upper).." "..(slot == "right" and 2 or slot == "left" and 1) }) } end,
```
Comment (Wires77) [inline Modules/ModParser.lua:2359]: I was originally trying to make the "you are hexproof" bit a separate thing, but it required more code than I was ready to implement, so this line is needlessly generic.  I might take another crack at it if this doesn't get merged before I find the time

Comment (ppoelzl) [reply Modules/ModParser.lua:2359]: Let's take the time to use the right abstraction instead.

**Comment by Wires77** [comment]
I've made this more generic, but I'm realizing there are a lot of mods that are basically flags that aren't technically flags.  Hexproof and stun immune are the first ones that stand out.  There should be a separate table for special mods that can still use the preFlags and modTags to augment them.  That's a separate PR though, I don't want to make this one more complicated than it needs to be.

**Review by ppoelzl — APPROVED** [review]
Looks reasonably fine. I hope we can clean this up in the future though.

**Pull request merged**